shuffler/logic/Dungeons/Fire Temple/FireTemple.logic

```diff
@@ -22,7 +24,10 @@ area FireTemple:
     node StairsMiddle:
       door F2Middle
 
-    Entrance <-> TopRight: open FireKey
+    Entrance -> LockedDoorToTopRight
+    TopRight -> LockedDoorToTopRight
+    LockedDoorToTopRight -> Entrance: open LockedDoorToTopRight
+    LockedDoorToTopRight -> TopRight: open LockedDoorToTopRight
```

Comment:
@Redriel57 Made this change because `Entrance <-> TopRight: open FireKey` doesn't properly encode the key door into the logic. The semantics of the `lock` and `open` descriptors are as follows:
Given `lock Name` on a node, `open Name` on an edge, `open A` means you can access the node with `lock A` and open it.